test/simplefootball_web_web/controllers/competition_controller_test.exs

```diff
@@ -36,6 +36,121 @@ defmodule SimplefootballWebWeb.CompetitionControllerTest do
                "competitionType" => "bundesliga2"
              }
            ]
+
+    Repo.delete_all(Competition)
+  end
+
+  test "get an error because no current matchday is existing", %{conn: conn} do
+    {:ok, bundesliga} =
+      createCompetition(%Competition{
+        name: "1. Bundesliga",
+        competition_type: "bundesliga",
+        icon_url:
+          "https://upload.wikimedia.org/wikipedia/en/thumb/d/df/Bundesliga_logo_%282017%29.svg/290px-Bundesliga_logo_%282017%29.svg.png"
+      })
+
+    {:ok, bundesliga2018} =
+      Repo.insert(%Season{
+        year: 2018,
+        competition_id: bundesliga.id
+      })
+
+    {:ok, _} =
+      Repo.insert(%Matchday{
+        number: 24,
+        description: "24. Spieltag",
+        season_id: bundesliga2018.id,
+        is_current_matchday: false
+      })
+
+    conn =
+      get(conn, Routes.competition_path(conn, :current_matchday, bundesliga.competition_type))
+
+    assert response(conn, 422) == "No current matchday"
+
+    Repo.delete_all(Matchday)
+    Repo.delete_all(Season)
+    Repo.delete_all(Competition)
+  end
+
+  test "get current matchday with existing one", %{conn: conn} do
+    date = DateTime.truncate(DateTime.utc_now(), :second)
```

Comment:
Dieser Test ist abhängig von der aktuellen Zeit (und selbst wenn er das tatsächlich nicht ist: Diese Zeile lässt mich das zumindest *denken*). Das ist schlecht. In einer perfekten Welt ist Code nie abhängig von so etwas externem wie "now".

Jetzt komm ich mal wieder :D Das Kapitel "Introduce Business Time" aus "Fifty Quick Ideas to Improve your Tests" ist interessant: https://drive.google.com/drive/u/0/folders/1nQQkjHqdyR141tdotll8x7ydYIo2dN9G

Konkret würde ich hier in meinem Tests lieber hard-codierte Zeiten nutzen als "now"